docs/design/fabric-v2+/fpc-registration.puml

```diff
@@ -12,261 +12,155 @@
-  FPC chaincode has three different associated endorsement policies (EP):
-  - lifecycle transactions ("normal fabric": approve chaincode definition (define application transaction EP), upgrade, etc ...)
-  - application transactions
-  - FPC-lifecycle transactions (register, key-gen, key-dist, ...)
-
-  Decision: We use a **separate** ERCC.
-  We realized FPC-liflecycle as application transaction EP of ERCC chaincode;
-  should ideally match lifecycle EP which is 'Channel/Application/LifecycleEndorsement' and by default majority of organizations)
-  Note that with the **integrated** approach we have deal with a mix of applications transactions EP and FPC-lifecycle EP.
-end note
-
-ref over User1, ERCC1
-  Create channel; each peer joins channel; each peer launches TLCC, which will  "imprint" itself with CH_id (content-addressable channel identifier, hash of gensis block)
+ref over ERCC1, ERCC2
+  It is assumed that: channel has already been created; ERCC has been installed.
+  Also: ERCC contains hard-coded the TLCC_MRenclave and the IAS Root Certificate.
 end ref
-|||
 
-ref over User1, ERCC1
-  Install and approve ERCC chaincode. ERCC defines TLCC_MR and IAS_PK in chaincode definition.
-  (alternatively, TLCC_MR and IAS_PK could be in channel configuration; crucial is that they are approved by humans (admins) on sufficiently many organizations)
-  ERCC also reads CH_id from ledger or, if that does not work, also embeds it into chaincode
-
-  see fpc-lifecycle-v2.puml
-end ref
-|||
-
-
-group enclave creation and attestation generation
-  
-  activate Admin
-
-  Admin -> Manager1 ++: create (CC_NAME, CC_VER, CH_name)
-  note right
-    CC_NAME: chaincode name
-    CC_VER: chaincode version
-    CH_NAME: channel name (needed by the ECC_Manager to cc2cc ECC
-  end note
-  note over Manager1 #FFaa00
+group enclave creation
+  note over Admin #FFaa00
     **MST**: we might want to replace (here and elsewhere) CC_VER with the sequence
     number from chain code definition as versions can be re-used in downgrades yet we
     might or might not want to re-use enclaves?
   end note
 
-  Manager1 -> Manager1 : get SPID and IAS credentials
-  note right: from local peer config
-
-  Manager1 -> ECC1 ++: create (SPID)
-  note left: this is a cc2cc call
-
-
-  ECC1 -> TLCC1 ++: getInfo
-  note over TLCC1
-    This information could also be retrieved elsewhere if easier. At this point in time
-    it does not have to be authenticated (and is anyway passed through untrusted ECC),
-    consistency will be verified later in registration (Remember that TLCC_MR is defined
-    and agreed by human consensus in the ERCC chaincode definition)
-  end note
-  return TLCC_MR, CH_id
-
-
-  |||
-
-  create participant "ECC_Enclave" as Enclave1 order 30 #99FF99
-  ECC1 -> Enclave1 ++: create (SPID, CC_NAME, CC_VER, Org_1, TLCC_MR, CH_id)
-  note left
-    this is a cgo call
-    Org_1 denotes the peer's MSP_ID
-  end note
+  Admin -> ECC1 ++: invoke create(SPID)
```

Comment:
as discussed in phoenix, we probably should either separate user action from client (code) function or just have external triggering events to characterize what the end-user has to do.  E.g., there doesn't seem any reason why would want to have the human do the later ias interaction and ercc registration steps.

Reply:
any suggestion on how to do the separation?
I have added a comment the these operations can be scripted/automated and that human intervention is sufficient but not necessary.

Reply:
the cleanest would be to separate the admin/client _human_ and _sw_ as two seperate entitites.  An easier (though maybe visually more confusing) would be to add the human triggering the action as a note (ideally with a different color as the other ones)

Reply:
I have tried to give it a shot